### ISSUE TITLE
Project Feedback

### ISSUE THREAD

**Issue by maliklau** [maliklau/MI-449-SS17-741-js-server-intro-jtrO3q#1]
@stuartpearman Can you take a look at this? It's [hosted here](https://maliklau-kappab1hhmo94jtro3q.herokuapp.com/) and meets the following criteria:

- [x] At every URL, you respond with HTML, starting with an `h1` element (though for this project, it doesn't have to be a complete HTML page)
- [x] When a user visits `/`, they see a greeting
- [x] When a user visits `/random-joke`, they see one of at least three [knock-knock jokes](http://www.short-funny.com/best-knock-jokes.php) (`Math.random()` may be useful 😉)
- [x] When a user visits `/cuteness`, they see a picture of a cute animal (I'm sure you can find one on the Internet 😛)
- [x] At every other URL, display an error page (if you want to make it funny, see [these examples](http://www.hongkiat.com/blog/60-really-cool-and-creative-error-404-pages/) for inspiration)
- [x] The error page includes the requested URL, similar to [Google's error page](http://google.com/secret-plan-to-take-over-the-world)
- [x] Every page that is _not_ the homepage (`/`) contains a link to it

**Comment by maliklau** [comment]
I know I turned it in, but I'm not sure the random jokes is working please let me know if I need to fix it/I was not sure why it wasn't working. I tried seeding my other random  number in the hopes that it would give me a different number. thanks!

**Comment by stuartpearman** [comment]
Yep, everything works except the random joke. The problem is that you're creating 2 different random numbers:

```
randInt = parseInt(Math.random())
randInt2 = parseInt(Math.random(500))
```

Both are actually going to be `0` everytime. In javascript, there are _sort of_ two types of numbers: integers and floats, although both are of the type "number". You've probably used floats and ints in other programming classes, but integers are only whole numbers like 0, 1, 2, etc. whereas floats can have decimal values as well.

The `parseInt` function basically cuts off everything to the right of the decimal everytime, leaving you with `0`. You could change this to `parseFloat`, but you would still probably only ever get 2 of the 3 jokes, because very rarely would the two number be equal. I suggest using one random number and determining the joke based on that value, rather than comparing two.

Let me know if you have any questions :+1:

**Comment by maliklau** [comment]
So I am trying to use just the top randomInt now, and am comparing it to the integer 50 for each joke, differentiating the if conditions by using comparisons === > <. It still doesn't seem to be working. Is that what you meant by using one random number for the joke? Like what should I be comparing it to? Thanks!

**Comment by stuartpearman** [comment]
`Math.random()` returns a number between 0 and 1, and `parseInt` rounds down to a whole number so you could do `parseInt(math.random() * 100)`. About half the time it would be below 50, about half the time it would be above, and about 1 in 100 times it would equal 50. I would recommend setting it up so all jokes have a roughly equal chance of appearing.

**Comment by maliklau** [comment]
okay I think I fixed it! Thanks! :)

**Comment by stuartpearman** [comment]
It seems to work when I pull down your code, but not so well on the hosted site, did you push everything to heroku?

**Comment by maliklau** [comment]
okay! I pushed it to Heroku, that was my bad I forgot. Hopefully it is all good now :) 
Thanks

**Comment by stuartpearman** [comment]
🎉 💯 :shipit:
